### PR TITLE
User Story 30 User Can Cancel Order

### PR DESCRIPTION
**Pull request by steve-meyers**
- Add ItemOrder controller
- Add Update functionality to ItemOrder controller
- Add helper methods to ItemOrder controller: cancel_order, unfulfill_item_orders
- Add enum to ItemOrder model
- Add patch route for ItemOrder update
- [x] Passes all tests?

### CODE REVIEW

File: app/models/item_order.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class ItemOrder <ApplicationRecord
   belongs_to :item
   belongs_to :order
 
+  enum status: %w(unfulfilled fulfilled)
```
Comment (jrsewell400) [inline app/models/item_order.rb:7]: nice use of enum here

File: app/controllers/item_orders_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class ItemOrdersController < ApplicationController
+  def update
+    order = Order.find(params[:id])
+    cancel_order(order)
+  end
+  
+  private
```
Comment (jrsewell400) [inline app/controllers/item_orders_controller.rb:7]: Curious what making these methods private does?

**Review by jrsewell400 — APPROVED** [review]
Testing and functionality additions look good from what I can see, great work guys!

File: app/controllers/item_orders_controller.rb (hunk 2 above)
Comment (zachholcomb) [reply app/controllers/item_orders_controller.rb:7]: Yeah i thought that was the convention, not really sure other than that, should they not be?

Comment (steve-meyers) [reply app/controllers/item_orders_controller.rb:7]: Making them private just means that only this controller can use these methods, right? But that would be the case no matter what I supposed

**Pull request merged**